katharsis-core/src/main/java/io/katharsis/resource/registry/repository/adapter/ResourceRepositoryAdapter.java

```diff
@@ -0,0 +1,189 @@
+
+	private final Object resourceRepository;
+
+	private final boolean isAnnotated;
+
+	public ResourceRepositoryAdapter(ResourceInformation resourceInformation, ModuleRegistry moduleRegistry,
+			Object resourceRepository) {
+		super(resourceInformation, moduleRegistry);
+		this.resourceRepository = resourceRepository;
+		this.isAnnotated = resourceRepository instanceof AnnotatedResourceRepositoryAdapter;
+	}
+
+	public JsonApiResponse findOne(I id, QueryAdapter queryAdapter) {
+		RepositoryRequestFilterChainImpl chain = new RepositoryRequestFilterChainImpl() {
+
+			@SuppressWarnings("rawtypes")
+			@Override
+			protected JsonApiResponse invoke(RepositoryFilterContext context) {
+				RepositoryRequestSpec request = context.getRequest();
+				QueryAdapter queryAdapter = request.getQueryAdapter();
+				Serializable id = request.getId();
+				Object resource;
+				if (isAnnotated) {
+					resource = ((AnnotatedResourceRepositoryAdapter) resourceRepository).findOne(id, queryAdapter);
+				}
+				else if (resourceRepository instanceof QuerySpecResourceRepository) {
+					resource = ((QuerySpecResourceRepository) resourceRepository).findOne(id,
+							request.getQuerySpec(resourceInformation.getResourceClass()));
+				}
+				else {
+					resource = ((ResourceRepository) resourceRepository).findOne(id, request.getQueryParams());
+				}
+				return getResponse(resourceRepository, resource, request);
+			}
+
+		};
+		RepositoryRequestSpec requestSpec = RepositoryRequestSpecImpl.forFindId(moduleRegistry.getResourceRegistry(),
+				queryAdapter, id);
+		return chain.doFilter(newRepositoryFilterContext(requestSpec));
+	}
+
+	public JsonApiResponse findAll(QueryAdapter queryAdapter) {
+		RepositoryRequestFilterChainImpl chain = new RepositoryRequestFilterChainImpl() {
+
+			@SuppressWarnings("rawtypes")
+			@Override
+			protected JsonApiResponse invoke(RepositoryFilterContext context) {
+				RepositoryRequestSpec request = context.getRequest();
+				QueryAdapter queryAdapter = request.getQueryAdapter();
+				Object resources;
+				if (isAnnotated) {
+					resources = ((AnnotatedResourceRepositoryAdapter) resourceRepository).findAll(queryAdapter);
+				}
+				else if (resourceRepository instanceof QuerySpecResourceRepository) {
+					resources = ((QuerySpecResourceRepository) resourceRepository)
+							.findAll(request.getQuerySpec(resourceInformation.getResourceClass()));
+				}
+				else {
+					resources = ((ResourceRepository) resourceRepository).findAll(request.getQueryParams());
+				}
+				return getResponse(resourceRepository, resources, request);
+			}
+
+		};
+		RepositoryRequestSpec requestSpec = RepositoryRequestSpecImpl.forFindAll(moduleRegistry.getResourceRegistry(),
+				queryAdapter);
+		return chain.doFilter(newRepositoryFilterContext(requestSpec));
+	}
+
+	public JsonApiResponse findAll(Iterable ids, QueryAdapter queryAdapter) {
+		RepositoryRequestFilterChainImpl chain = new RepositoryRequestFilterChainImpl() {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Reduce this anonymous class number of lines from 23 to at most 20, or make it a named class. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1188)